### PR TITLE
Various small improvements

### PR DESCRIPTION
**Pull request by basvanbeek**
- Added WithMaxLogsPerSpan tracer option
- Added LogFmt and ZipkinStrict materializers for OpenTracing KV Logs to Zipkin Annotations.

### CODE REVIEW

File: log-materializers_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		t.Fatalf("expected json string, got error %+v", err)
+	}
+	if want != string(have) {
+		t.Errorf("want:\n%s\nhave\n%s", want, have)
+	}
+}
+
+func TestMaterializeWithLogFmt(t *testing.T) {
+	logFields := getLogFields()
+	want := `bool=true string=value error="an error" float32=32.123 float64=64.123 int=42 int32=32 int64=64 uint32=32 uint64=64 object="unsupported value type" event=EventValue`
```
Comment (adriancole) [inline log-materializers_test.go]: cc @lookfwd this format looks much better in the zipkin UI although in general it is preferred to use a simple string to annotate an important moment in the span.

**Comment by basvanbeek** [comment]
Thanks for the comments @adriancole. 

Baking this to a release as we now properly flush transporter queues thanks to @richardmarshall and we have a better way to deal with OpenTracing KV Logs.

File: log-materializers_test.go (hunk 1 above)
Comment (lookfwd) [inline log-materializers_test.go]: Nice! Is there any specific place in zipkin code where the parsing of any annotation that is good-looking happens?

Comment (basvanbeek) [inline log-materializers_test.go]: You can select the materializer you want to use for OT Logs to Zipkin Annotations when wiring up the Recorder with `NewRecorder()`. Default is LogFmt but you can change to:

```go
# Zipkin Strict Mode (only materialize the value from an "event" Log Field):
NewRecorder(collector, debug, hostPort, serviceName, WithStrictMaterializer())

# JSON formatted
NewRecorder(collector, debug, hostPort, serviceName, WithJSONMaterializer())

# Explicit LogFmt
NewRecorder(collector, debug, hostPort, serviceName, WithLogFmtMaterializer())

```

**Pull request merged**